packages/components/file-card/src/file-card.js

```diff
@@ -0,0 +1,357 @@
+														id="Icon"
+													></path>
+												</g>
+											</g>
+										</g>
+									</g>
+								</g>
+							</g>
+						</svg>
+					</span>
+				</div>
+			`;
+		}
+
+		get fileInformation() {
+			if (this.state === 'download') {
+				const fileExtension = this.fileObject.name
+					.split('.')
+					.pop()
+					.toUpperCase();
+
+				return html`
+					<div class="file-information">
+						<ts-file-size size="${this.fileObject.size}"></ts-file-size>
+						<ts-typography
+							text="${`| ${fileExtension} Document`}"
+							variant="subtitle"
+						></ts-typography>
+					</div>
+				`;
+			}
+		}
+
+		get actionMessage() {
+			return html`
+				<ts-typography
+					class="remove-action-message"
+					text="Remove"
+					color="action"
+					variant="subtitle"
+				></ts-typography>
+				<ts-typography
+					class="download-action-message"
+					text="Download"
+					color="action"
+					variant="subtitle"
+				></ts-typography>
+			`;
+		}
+
+		render() {
+			return html`
+				<div
+					?data-removable="${this.removable}"
+					class="file-card-wrapper state-${this.state}"
+				>
+					<ts-card
+						type="${this.cardType}"
+						orientation="horizontal"
+						?rtl="${this.rtl}"
+						data-size="${this.size}"
+					>
+						${this.fileIcon}
+						<div class="content">
+							<ts-typography
+								text="${this.fileObject.name}"
+								class="file-name"
+								variant="title"
+								no-wrap
+							></ts-typography>
+							${this.progressBar} ${this.fileInformation} ${this.errorMessages}
+							${this.actionMessage}
+						</div>
+						${this.actionIcons}
+					</ts-card>
+				</div>
+			`;
+		}
+
+		firstUpdated(_changedProperties) {
+			this.handleRemoveActionEvents();
+			this.handleDownloadActionEvents();
+		}
+
+		connectedCallback() {
+			super.connectedCallback();
+			document.addEventListener('readystatechange', this.handleChange);
+		}
+
+		disconnectedCallback() {
+			document.removeEventListener('readystatechange', this.handleChange);
+			super.disconnectedCallback();
+		}
+
+		removeFile() {
+			if (this.removable) {
+				let event = new CustomEvent('remove', {
+					detail: {
+						file: this.fileObject
+					},
+					bubbles: true,
+					composed: true
+				});
+
+				this.dispatchEvent(event);
+			}
+		}
+
+		downloadFile() {
+			if (this.state === 'download') {
+				let event = new CustomEvent('remove', {
+					detail: {
+						file: this.fileObject
+					},
+					bubbles: true,
+					composed: true
+				});
+
+				this.dispatchEvent(event);
+			}
+		}
+
+		toggleRemoveActionHoverState(event) {
+			this.shadowRoot
+				.querySelector('.file-card-wrapper')
+				.classList.toggle('remove-action-hovered');
+		}
+
+		toggleDownloadActionHoverState(event) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "event". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)